lib/TH/generic/THVectorDispatch.c

```diff
@@ -32,7 +31,7 @@ void THVector_(fill)(real *x, const real c, const ptrdiff_t n) {
   THVector_(fill_DISPATCHPTR)(x, c, n);
 }
 
-
+#ifndef TH_GENERIC_NO_MATH
```

Comment:
Probably should be removed as well. I thought enabling initialization via fill()was a good idea.